src/containers/ArticlePage/FrontpageArticlePage/components/FrontpageArticleProvider.tsx

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2023-present, NDLA.
+ *
+ * This source code is licensed under the GPLv3 license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+import { createContext, Dispatch, ReactNode, SetStateAction, useContext, useState } from 'react';
+
+const FrontpageArticleContext = createContext<[boolean, Dispatch<SetStateAction<boolean>>]>([
+  false,
+  (val) => val,
+]);
+
+interface Props {
+  children: ReactNode;
+  initialValue?: boolean;
+}
+
+export const isFrontpageArticle = (draftId: number) => getArticleIdList().includes(draftId);
+
+export const FrontpageArticleProvider = ({ children, initialValue = false }: Props) => {
+  const isFrontpageArticle = useState<boolean>(initialValue);
+  return (
+    <FrontpageArticleContext.Provider value={isFrontpageArticle}>
+      {children}
+    </FrontpageArticleContext.Provider>
+  );
+};
+
+export const useFrontpageArticle = () => {
+  const context = useContext(FrontpageArticleContext);
+
+  const [isFrontpageArticle, setFrontpageArticle] = context;
+
+  const toggleFrontpageArticle = (articleId: number) => {
+    const frontpageArticleIds = getArticleIdList();
+
+    if (frontpageArticleIds.includes(articleId)) {
+      updateFrontpageArticleList(frontpageArticleIds.filter((value) => value !== articleId));
+      setFrontpageArticle(false);
+    } else {
+      updateFrontpageArticleList([...frontpageArticleIds, articleId]);
+      setFrontpageArticle(true);
+    }
+  };
+
+  return {
+    isFrontpageArticle,
+    toggleFrontpageArticle,
+    setFrontpageArticle,
+  };
+};
+
+const getArticleIdList: () => number[] = () =>
+  JSON.parse(localStorage.getItem('frontpage-articles') ?? '[]')
+    .filter(Number)
+    .map(Number);
+
+const updateFrontpageArticleList = (articles: number[]) =>
+  localStorage.setItem('frontpage-articles', JSON.stringify(articles));
```

Comment:
Henger ikke helt med her. Vi kan vel bare ha en forside?

Reply:
Ser også for meg at vi kommer til å få uendelig med bug-reports dersom dette skal lagres i localStorage, og være forskjellige hos alle.

Reply:
Alle om-ndla-artikler kan potensiellt settes som forside. Du må kunne lage/redigere en artikkel til forsida og velge å setten den til brei versjon for å kunne sjå den korrekt.